alphagov/paas-cf
# [#111509456] - Lint checks for shell scripts

## What

Use shellcheck to check all our shell scripts for common errors. This also fixes all the errors that were reported.

## Details

This includes some refactorings:

The `bosh-manitest/build_manifest.sh` has been refactored to be more consistent
with the cloudfoundry one and take the paths to output files in environment
variables. This was done to avoid some issues with array variables in POSIX
scripts that were being flagged up by shellcheck (bash isn't available in
alpine based containers). See 9138c37

The travis build has been refactord to be a single job instead of a matrix
build as it's more efficient this way. This also included creating a
Makefile which will make it easier to run locally. (see 2538f91)

## How to test

Run the tests localy - `make test` (these will require shellcheck > 0.4.0 to be
installed locally).

On Mac - `brew install shellcheck`

On Linux:
   git clone git@github.com:koalaman/shellcheck.git
   sudo apt-get install cabal-install ghc
   cd shellcheck
   cabal update
   cabal install # installs into ~/.cabal/bin

Confirm that the deployment process still works after the various refactorings.

## Who can review

Anyone but @alext  or myself.

##Notes : We can re-enable the SC2086 checks in the `s3init.sh` and `s3get.sh` scripts after Shellcheck 0.4.3 is available for everyone.
